src/compilers/compile-js.js

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright the eleventy-plugin-fluid copyright holders.
+
+See the AUTHORS.md file at the top-level directory of this distribution and at
+https://github.com/fluid-project/eleventy-plugin-fluid/raw/main/AUTHORS.md.
+
+Licensed under the New BSD license. You may not use this file except in compliance with this License.
+
+You may obtain a copy of the New BSD License at
+https://github.com/fluid-project/eleventy-plugin-fluid/raw/main/LICENSE.md.
+*/
+"use strict";
+
+const esbuild = require("esbuild");
+const path = require("node:path");
+
+module.exports = async (content, inputPath, options) => {
+    let parsed = path.parse(inputPath);
+    if (!inputPath.startsWith(options.basePath) || parsed.name.startsWith("_")) {
+        return;
+    }
+
+    return async () => {
+        let esbuildOptions = options;
```

Comment:
I don't think this will make a new object but will just assign the reference of `options` to `esbuildOptions`. You could use the spread operator or Object.assign to make a copy.
```suggestion
        let esbuildOptions = {...options};
```